src/appleseed.python/bindblenderprogressivetilecallback.cpp

```diff
@@ -173,32 +181,110 @@ namespace
                     m_updated_buffer = false;
                 }
 
-                glColor4f(1.0f, 1.0f, 1.0f, 1.0f);
+                if (!m_updated_data_buffer)
+                {
 
-                glEnable(GL_TEXTURE_2D);
+                    m_indices =
+                    {
+                        0, 1, 3,
+                        1, 2, 3
+                    };
+
+                    m_texture_coords =
+                    {
+                        0.0f, 1.0f,
+                        1.0f, 1.0f,
+                        1.0f, 0.0f,
+                        0.0f, 0.0f
+                    };
+
+                    m_vertex_coords =
+                    {
+                        -1, -1,
+                         1, -1,
+                         1,  1,
+                        -1,  1
+                    };
+
+                    // Get shader program set by blender.
+                    glGetIntegerv(GL_CURRENT_PROGRAM, &m_shader_program_id);
+
+                    // Create vertex shader if there isn't one.
+                    GLint shader_count;
+                    glGetProgramiv(m_shader_program_id, GL_ATTACHED_SHADERS, &shader_count);
+                    if (shader_count < 2)
```

Comment:
How do you know that if shader_count < 2, then it's the vertex shader that's missing? And how do you know that if shader_count >= 2, then both the fragment and vertex shaders are there?

Reply:
Basically, Blender 2.79 create just a fragment shader, while blender 2.8 create the frag and vertex shaders.
The GLSL shaders create by blender:
[gpu_shader_display_transform_Blender_27.txt](https://github.com/appleseedhq/appleseed/files/3077737/gpu_shader_display_transform_Blender_27.txt)
[gpu_shader_display_transform_Blender_28.txt](https://github.com/appleseedhq/appleseed/files/3077738/gpu_shader_display_transform_Blender_28.txt)
[gpu_shader_display_transform_vertex_Blender_28.txt](https://github.com/appleseedhq/appleseed/files/3077739/gpu_shader_display_transform_vertex_Blender_28.txt)

Reply:
They are created here: https://developer.blender.org/diffusion/B/browse/master/intern/opencolorio/ocio_impl_glsl.cc$284
for 2.8
and here: https://developer.blender.org/diffusion/B/browse/blender2.7/intern/opencolorio/ocio_impl_glsl.cc$254.
for 2.7.
At the function setupGLSLDraw.

Reply:
Thanks for the explanations and for digging up the relevant source code!

Reply:
It would be nice to add a comment with a short explanation of this.